src/TreeSelect.tsx

```diff
@@ -11,9 +11,12 @@ export const TreeSelect = ({ name, validate, ...restProps }: TreeSelectProps) =>
     {({ field: { value }, form }: FieldProps) => (
       <$TreeSelect
         value={value}
-        onChange={e => form.setFieldValue(name, e.valueOf())}
-        onBlur={e => form.setFieldTouched(name)}
         {...restProps}
```

Comment:
Currently {...restProps} should come after setting the setting onChange onBluer value etc, to allow to override them.

This might not be good in general. Exposing onChange handlers might make this approach obsolete. However we should not change the current behavior.

Pls make {...restProps} the last "prop"

Reply:
I fixed like #58

Reply:
Yes I see.
However to have a clean history and be able to revert easily its better to have separate PRs/commits for different topics.

Reply:
You're right